File: frontend/src/types/examen.ts
export interface ExamenMetadata {
  patient_id: string
  study_instance_uid: string
  date_examen: string | null
  nb_coupes: number
  dimensions: number[] | null
  pixel_spacing: number[] | null
  slice_thickness: number | null
  demo?: boolean
}

export interface UploadResponse {
  task_id: string | null
  study_id: string | null
  nb_coupes: number | null
  metadata: ExamenMetadata | null
  preview_slices: number[]
  files_received: number
  total_files: number | null
  finalized: boolean
}

export interface UploadStatus {
  task_id: string
  status: 'pending' | 'running' | 'done' | 'error'
  progress: number
  files_received: number
  total_files: number | null
  error: string | null
  result: UploadResponse | null
}

export interface Examen {
  id: number
  study_instance_uid: string
  patient_id: string
  date_examen: string | null
  nb_coupes: number
  dicom_path: string
  uploaded_at: string
  uploaded_by: number
}

export type ExamenResultFilter = 'all' | 'fracture' | 'normal'
export type ExamenPeriodFilter = 'all' | 'week' | 'month'

export interface ExamenListItem {
  id: number
  study_id: string
  patient_id: string
  date: string
  nb_coupes: number
  uploaded_at: string
  vertebres: string[]
  score_global: number | null
  fracture_detectee: boolean | null
  analysed: boolean
  uploaded_by?: number | null
  medecin_nom?: string | null
}

export interface ExamenListResponse {
  items: ExamenListItem[]
  total: number
  page: number
  limit: number
}

export interface FetchExamensParams {
  page?: number
  limit?: number
  search?: string
  result?: ExamenResultFilter
  period?: ExamenPeriodFilter
}

File: backend/app/schemas/examen.py
from datetime import datetime

from pydantic import BaseModel, Field


class ExamenMetadataSchema(BaseModel):
    patient_id: str
    study_instance_uid: str
    date_examen: str | None = None
    nb_coupes: int
    dimensions: list[int] | None = None
    pixel_spacing: list[float] | None = None
    slice_thickness: float | None = None
    demo: bool = False


class UploadResponseSchema(BaseModel):
    task_id: str | None = None
    study_id: str | None = None
    nb_coupes: int | None = None
    metadata: ExamenMetadataSchema | None = None
    preview_slices: list[int] = Field(default_factory=list)
    files_received: int = 0
    total_files: int | None = None
    finalized: bool = False


class UploadStatusSchema(BaseModel):
    task_id: str
    status: str
    progress: int
    files_received: int
    total_files: int | None = None
    error: str | None = None
    result: UploadResponseSchema | None = None


class ExamenSchema(BaseModel):
    id: int
    study_instance_uid: str
    patient_id: str
    date_examen: datetime | None
    nb_coupes: int
    dicom_path: str
    uploaded_at: datetime
    uploaded_by: int

    model_config = {"from_attributes": True}


class ExamenListItemSchema(BaseModel):
    id: int
    study_id: str
    patient_id: str
    date: datetime
    nb_coupes: int
    uploaded_at: datetime
    vertebres: list[str] = Field(default_factory=list)
    score_global: float | None = None
    fracture_detectee: bool | None = None
    analysed: bool = False
    uploaded_by: int | None = None
    medecin_nom: str | None = None


class ExamenListResponse(BaseModel):
    items: list[ExamenListItemSchema]
    total: int
    page: int
    limit: int
